### PR TITLE
use delete_symLink only if it's neccessary

### LINKED ISSUE #4622: Delete symlink script is called on closing tasks
The script `script_deleteSymLink.sh` get called every time a task is closed. This is only necessary if the task property `write images` is set as then on taking over this task the script `script_createSymLink.sh` is called and the symbolic link between the images directory of the process into the user home directory is done

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
fixes #4622

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java [hunk 1]
```diff
@@ -211,15 +211,13 @@ public void closeTaskByUser(Task task) throws DataException, IOException, DAOExc
                 if (!mih.checkIfImagesValid(task.getProcess().getTitle(), imageFolder)) {
                     throw new DataException("Error on image validation!");
                 }
+                this.webDav.uploadFromHome(task.getProcess());
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java:214]: This move is wrong as the method is only called when the task should be verified on closing and the images could be modified.

I would recommend to 
- not move the call to this place
-  on original place wrap the call with a test to `task.isTypeImagesRead() || task.isTypeImagesWrite()`

File: Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java [hunk 2]
```diff
@@ -2419,15 +2419,12 @@ private static void deleteMetadataDirectory(Process process) {
...
-        WebDav webDav = new WebDav();
-
-        for (Role role : task.getRoles()) {
-            for (User user : role.getUsers()) {
-                try {
-                    webDav.uploadFromHome(user, task.getProcess());
-                } catch (RuntimeException e) {
-                    Helper.setErrorMessage(e.getLocalizedMessage(), logger, e);
-                }
+        if (task.isTypeImagesWrite()) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java:2422]: The test must be extended to a check if `task.isTypeImagesRead()` as even there a symlink is created but only a read-only symlink.

**Review by henning-gerhardt — CHANGES_REQUESTED** [review]
There must be at least one more (or maybe more) calls to `webDav.uploadFromHome()` which are executed independent if the task has isTypeImagesRead or isTypeImagesWrite.

File: Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java (hunk 1 above)
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java:214]: By the way, @Kathrin-Huber, could you maybe rename the method? Nothing is uploaded there. No idea how, but that name doesn't make any sense.

Comment (henning-gerhardt) [reply Kitodo/src/main/java/org/kitodo/production/services/workflow/WorkflowControllerService.java:214]: Not only the method even the whole class WebDav did not make this what the name is suggesting. Renaming should be done after this pull request is merged.

**Pull request merged**